### ISSUE TITLE
Unable to dump source

### ISSUE THREAD

**Issue by trescube** [openaddresses/pyesridump#35]
Trying to query for `https://gismaps.sedgwickcounty.org/arcgis/rest/services/Map/Op_SiteAddress_Dynamic_SP/MapServer/0` returns this error:

```
2017-03-01 11:44:36,011 - cli.esridump - ERROR - Could not parse response from https://gismaps.sedgwickcounty.org/arcgis/rest/services/Map/Op_SiteAddress_Dynamic_SP/MapServer/0/query?returnCountOnly=true&where=1%3D1&f=json as JSON:

<html><head><title>Request Rejected</title></head><body>The requested URL was rejected. Please consult with your administrator.<br><br>Your support ID is: 11833783245905056836</body></html>
```

How does one go about dumping a source that is locked down tightly like this one?  Is it possible to do with pyesridump as-is?  If so, could documentation be added so people who aren't well-versed in arcgis/esri/whatever-term can try different approaches to dealing with problematic servers (this is my situation)?

**Comment by iandees** [comment]
It looks like their firewall/proxy is misconfigured. [Their webmap](https://gismaps.sedgwickcounty.org/portal/) has query functionality (that [makes a similar request as us](https://gismaps.sedgwickcounty.org/arcgis/rest/services/Map/Op_SiteAddress_Dynamic_SP/MapServer/0/query?inSR=102678&outSR=102678&geometryType=esriGeometryEnvelope&returnDistinctValues=false&geometry=%7B%22ymin%22%3A1709040%2E7800177576%2C%22xmin%22%3A1649095%2E792021425%2C%22xmax%22%3A1649352%2E4467612193%2C%22ymax%22%3A1709273%2E9243997082%7D&outFields=LABEL%2CMUNI%2CPOSTCO%2CZIP%2CLAT%2CLONG%2CMAPPEDTO%2CMULT%5FADD%2COBJECTID&spatialRel=esriSpatialRelIntersects&returnGeometry=true&f=amf)) that is failing right now because of this error you're seeing with pyesridump:

![image](https://cloud.githubusercontent.com/assets/261584/23473147/2ad46582-fe7d-11e6-88a4-431851b4869f.png)

**Comment by trescube** [comment]
Okay, regardless of misconfigured firewall/proxy on their part, is there any way right now that pyesridump can scrape the data?  I tried passing a custom `WHERE` but go the same thing:

```
$ esri2geojson -p "WHERE=OBJECTID > 1" https://gismaps.sedgwickcounty.org/arcgis/rest/services/Map/Op_SiteAddress_Dynamic_SP/MapServer/0 us-ks-sedgwick.geojson
2017-03-01 12:51:57,608 - cli.esridump - ERROR - Could not parse response from https://gismaps.sedgwickcounty.org/arcgis/rest/services/Map/Op_SiteAddress_Dynamic_SP/MapServer/0?WHERE=OBJECTID+%3E+1&f=json as JSON:

<html><head><title>Request Rejected</title></head><body>The requested URL was rejected. Please consult with your administrator.<br><br>Your support ID is: 11833783245905262104</body></html>
```

**Comment by iandees** [comment]
No, I don't think there's a way pyesridump can handle this server if it won't respond to the `/query` endpoint.

**Comment by iandees** [comment]
But you mentioned a brute force method... Did you try it on this layer and it worked?

**Comment by trescube** [comment]
I have seen this behavior before on Esri servers.  It's pretty rare but does happen.  

I can make this request as-is in both wget and chrome and get a JSON response:

`https://gismaps.sedgwickcounty.org/arcgis/rest/services/Map/Op_SiteAddress_Dynamic_SP/MapServer/0/1?f=json`

So I assume that by starting at 1 (or wherever) and just incrementing until the server returns a 400/404, the data can be scraped.

**Comment by iandees** [comment]
That will work for this layer because the first OID is `1` and the next one is `2`, etc., but in lots of other layers I look at the first OID is not 1 and there are gaps between the OIDs (this is why I do the OID enumeration method), so this won't be repeatable.

The other annoying thing with this is that the Esri server won't do the reprojection like we do via /query for you when you query by OID.

**Comment by trescube** [comment]
I've thought about ways to handle id gaps/offset but until a problematic source comes up, simple incrementing will do for now.
